src/include/homestore/replication/repl_dev.h

```diff
@@ -380,7 +379,7 @@ class ReplDevListener {
     }
 
     /// @brief ask upper layer to handle no_space_left event
-    virtual folly::Future< std::error_code > on_no_space_left(uint32_t pdev_id, chunk_num_t chunk_id) {
+    virtual folly::Future< std::error_code > on_no_space_left(chunk_num_t chunk_id) {
```

Comment:
most of the callback are sync.  
 Why we want to make this callback async(return a future) then waiting the future in sync manner?

Reply:
may be we can use thenValue to do something ? or we want to wait in another thread?  this left the possibility ^^

Reply:
If this is the argument we should move all CB into future:)

Probably this express your hidden intent that the `on_no_space_left` will take a long time.